### PR TITLE
NIFI-7087: Use FlowManager.findAllConnections() when available

### PR DESCRIPTION
**Pull request by mattyb149**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

Changed calls from StandardProcessGroup.findAllConnections() to FlowManager.findAllConnections() when a FlowManager object is available. Some places work specifically on a ProcessGroup, I did not change those calls, only the no-arg calls. I also changed some stream() calls into foreach loops for added performance.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on both JDK 8 and JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by YolandaMDavis** [comment]
@mattyb149 I tested these changes with a simple flow and predictions were generated as expected  across different levels of process groups. LGTM

**Comment by pvillard31** [comment]
I just gave it a try and it LGTM as well. @markap14 do you want to have a look or are you happy with me merging it into master?

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java [hunk 1]
```diff
@@ -4894,7 +4894,7 @@ public void verifyCanUpdate(final VersionedFlowSnapshot updatedFlow, final boole
                 // any connection that does exist in the proposed flow. This results in us having a Map whose values are those Connections
                 // that were removed. We can then check for any connections that have data in them. If any Connection is to be removed but
                 // has data, then we should throw an IllegalStateException.
-                findAllConnections().forEach(conn -> removedConnectionByVersionedId.put(conn.getVersionedComponentId().orElse(conn.getIdentifier()), conn));
+                flowManager.findAllConnections().forEach(conn -> removedConnectionByVersionedId.put(conn.getVersionedComponentId().orElse(conn.getIdentifier()), conn));
```
Comment (markap14) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java]: I think this call is actually changing the logic. `findAllConnections()` here is going to return all Connections in this ProcessGroup and any children/descendant groups. This changes it from "all connections here and below" to "all connections in the canvas," which I don't think is what we want.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java [hunk 2]
```diff
@@ -5018,9 +5018,11 @@ public void verifyCanUpdate(final VersionedFlowSnapshot updatedFlow, final boole
             final Map<String, VersionedConnection> proposedConnections = new HashMap<>();
             findAllConnections(updatedFlow.getFlowContents(), proposedConnections);
 
-            findAllConnections().stream()
-                    .filter(conn -> conn.getVersionedComponentId().isPresent())
-                    .forEach(conn -> proposedConnections.remove(conn.getVersionedComponentId().get()));
+            for (Connection conn : flowManager.findAllConnections()) {
```
Comment (markap14) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java]: Same comment as above. Here, I think we need to use the existing `findAllConnections()` method. We only want to change to `flowManager.findAllConnections()` when calling it on the root process group.

**Comment by mattyb149** [comment]
Sorry for the force-push override of the commit, but with the merge conflict it was just easier. All changes should be the same except now there are no changes to StandardProcessGroup

**Comment by markap14** [comment]
All looks good to me at this point +1

**Pull request closed without merging**